python/packages/azurefunctions/agent_framework_azurefunctions/_entities.py

```diff
@@ -131,6 +148,17 @@ async def run_agent(
                 type(agent_run_response).__name__,
             )
 
+            # Convert response into DurableAgentStateResponse and combine with request
+            state_response = DurableAgentStateResponse.from_run_response(correlation_id, agent_run_response)
+
+            # Get the pending request and combine its messages with the response messages
+            pending_request = self._pending_requests.pop(correlation_id, None)
+            if pending_request:
+                # Combine request and response messages into a single entry
+                state_response.messages = pending_request.messages + state_response.messages
+
+            self.state.data.conversation_history.append(state_response)
```

Comment:
Any reason why we're waiting for the response before adding the new user_message to the `conversation_history`? If something fails then we dont want to persist the last message? is that the use case?